src/aggregates/prom_increase.rs

```diff
@@ -1,109 +1,50 @@
-use pgx::error;
-use pgx::Internal;
 use pgx::*;
 
-use crate::aggregate_utils::in_aggregate_context;
 use crate::aggregates::{GapfillDeltaTransition, Milliseconds};
-use crate::palloc::{Inner, InternalAsValue, ToInternal};
 use crate::raw::TimestampTz;
 
-#[allow(clippy::too_many_arguments)]
-#[pg_extern(immutable, parallel_safe)]
-pub fn prom_increase_transition(
-    state: Internal,
-    lowest_time: TimestampTz,
-    greatest_time: TimestampTz,
-    step_size: Milliseconds, // `prev_now - step_size` is where the next window starts
-    range: Milliseconds,     // the size of a window to delta over
-    sample_time: TimestampTz,
-    sample_value: f64,
-    fc: pg_sys::FunctionCallInfo,
-) -> Internal {
-    prom_increase_transition_inner(
-        unsafe { state.to_inner() },
-        lowest_time.into(),
-        greatest_time.into(),
-        step_size,
-        range,
-        sample_time.into(),
-        sample_value,
-        fc,
-    )
-    .internal()
-}
-
-#[allow(clippy::too_many_arguments)]
-fn prom_increase_transition_inner(
-    state: Option<Inner<GapfillDeltaTransition>>,
-    lowest_time: pg_sys::TimestampTz,
-    greatest_time: pg_sys::TimestampTz,
-    step_size: Milliseconds, // `prev_now - step` is where the next window starts
-    range: Milliseconds,     // the size of a window to delta over
-    sample_time: pg_sys::TimestampTz,
-    sample_value: f64,
-    fc: pg_sys::FunctionCallInfo,
-) -> Option<Inner<GapfillDeltaTransition>> {
-    unsafe {
-        in_aggregate_context(fc, || {
-            if sample_time < lowest_time || sample_time > greatest_time {
-                error!("input time less than lowest time")
-            }
-
-            let mut state = state.unwrap_or_else(|| {
-                let state: Inner<_> = GapfillDeltaTransition::new(
-                    lowest_time,
-                    greatest_time,
-                    range,
-                    step_size,
-                    true,
-                    false,
-                )
-                .into();
-                state
-            });
-
-            state.add_data_point(sample_time, sample_value);
-
-            Some(state)
-        })
+pub struct PromIncrease {}
+
+#[pg_aggregate]
+impl Aggregate for PromIncrease {
+    type State = Option<GapfillDeltaTransition>;
+    type Args = (
```

Comment:
It's unfortunate that this trait requires an extra allocation just to pack the arguments into a tuple in order to call the `state` function, but I guess that was a tradeoff they were forced to make. I wonder what the impact of that is in a hot aggregate.